dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+    for (int64_t sample_idx = 0, flat_chunk_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = chunks_per_sample_[sample_idx].num_elements();
+      if (num_chunks == 0) {
+        continue;
+      }
+      int64_t cum_offset = 0;
+      auto sample_num_bytes = in_shape[sample_idx].num_elements();
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++, flat_chunk_idx++) {
+        offsets_.push_back(cum_offset);
+        cum_offset += sizes_[flat_chunk_idx];
+      }
+      DALI_ENFORCE(cum_offset <= sample_num_bytes,
+                   make_string("The sum of chunk sizes for sample of idx ", sample_idx,
+                               " exceeds the total size of the sample."));
+    }
+    assert(offsets_.size() == sizes_.size());
+  }
+
+  void SetupOutputShape(const TensorListView<StorageCPU, const int> &provided_shape) {
+    DALI_ENFORCE(provided_shape.sample_dim() == 0 || provided_shape.sample_dim() == 1,
+                 make_string("The shape argument must be a scalar or a 1D tensor, got tensor with ",
+                             provided_shape.sample_dim(), " extents."));
+    auto chunk_shapes = ParseOutputShape(provided_shape);
+    if (!HasChunks()) {
+      output_shape_ = chunk_shapes;
+    } else {
+      int batch_size = chunk_shapes.num_samples();
+      output_shape_.resize(batch_size, chunk_shapes.sample_dim() + 1);
+      for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+        int num_chunks = chunks_per_sample_[sample_idx].num_elements();
+        output_shape_.set_tensor_shape(
+            sample_idx, shape_cat(TensorShape<>{num_chunks}, chunk_shapes[sample_idx]));
+      }
+    }
+  }
+
+  TensorListShape<> ParseOutputShape(const TensorListView<StorageCPU, const int> &provided_shape) {
+    auto num_samples = provided_shape.num_samples();
+    if (num_samples == 0) {
+      return {};
+    }
+    int sample_dim = provided_shape[0].num_elements();
+    TensorListShape<> shape(num_samples, sample_dim);
+    for (int sample_idx = 0; sample_idx < provided_shape.num_samples(); sample_idx++) {
+      const int *data = provided_shape.tensor_data(sample_idx);
+      for (int d = 0; d < sample_dim; d++) {
+        DALI_ENFORCE(
+            data[d] >= 0,
+            make_string("Extents of the output shape must be non-negative integers. Please verify "
+                        "the sample of idx ",
+                        sample_idx, " passed as the argument `", shapeArgName, "`."));
+      }
+      TensorShape<> sample_shape(data, data + sample_dim);
+      max_output_sample_vol_ = std::max(max_output_sample_vol_, volume(sample_shape));
+      shape.set_tensor_shape(sample_idx, sample_shape);
+    }
+    return shape;
+  }
+
+  const OpSpec &spec_;
+
+  ArgValue<int, DynamicDimensions> shape_;
+  ArgValue<int, 1> chunk_offsets_;
+  ArgValue<int, 1> chunk_sizes_;
+  TensorLayout layout_;
+  TensorLayout sequence_extent_;
```

Comment:
We don't really need a tensor layout here - we need a single axis name.
```suggestion
  char sequence_axis_name_ = 0;
```

Reply:
done